workflow/controller/operator.go

```diff
@@ -1160,16 +1163,32 @@ func (woc *wfOperationCtx) checkParallelism(tmpl *wfv1.Template, node *wfv1.Node
 }
 
 func (woc *wfOperationCtx) executeContainer(nodeName string, tmpl *wfv1.Template, boundaryID string) *wfv1.NodeStatus {
+	var skipNodeInitialization bool
 	node := woc.getNodeByName(nodeName)
 	if node != nil {
-		return node
+		if strings.Contains(node.Message, exceededQuota) {
+			skipNodeInitialization = true
+			woc.log.Infof("Node %s exists but a pod does not: %s", nodeName, node.Message)
+		} else {
+			return node
+		}
 	}
 	woc.log.Debugf("Executing node %s with container template: %v\n", nodeName, tmpl)
+	var messages []string
 	_, err := woc.createWorkflowPod(nodeName, *tmpl.Container, tmpl)
 	if err != nil {
-		return woc.initializeNode(nodeName, wfv1.NodeTypePod, tmpl.Name, boundaryID, wfv1.NodeError, err.Error())
+		if strings.Contains(err.Error(), exceededQuota) {
+			messages = append(messages, exceededQuota)
+			woc.log.Infof("Failed to create pod due to a lack of resources. It will be marked pending: %v", err)
+		} else {
+			return woc.initializeNode(nodeName, wfv1.NodeTypePod, tmpl.Name, boundaryID, wfv1.NodeError, err.Error())
+		}
 	}
-	return woc.initializeNode(nodeName, wfv1.NodeTypePod, tmpl.Name, boundaryID, wfv1.NodePending)
+	if skipNodeInitialization {
```

Comment:
this branch is only reached when a node exists but the pod failed to create because of an `exceededQuota` error. Should a sleep be added here to avoid slamming the master with Pod create requests?